docs/user/fireworks.md

```diff
@@ -29,6 +29,14 @@ mgo_structure = Structure(
 # make a band structure flow to optimise the structure and obtain the band structure
 bandstructure_flow = RelaxBandStructureMaker().make(mgo_structure)
 
+# (Optional) add metadata to the flow task document.
+# Could be useful to filter specific results from the database.
+# For e.g., adding material project ID for the compound, use following lines
+bandstructure_flow.update_maker_kwargs(
+    {"_set": {"task_document_kwargs->additional_fields": {"mp_id": "mp-xxx"}}},
+    dict_mod=True,
+)
+
```

Comment:
Maybe I can learn something here. What are the virtues of `update_maker_kwargs` over something like this?

```
for job in flow if isinstance(flow, Flow) else [flow]:
    job.metadata[key] = val
```

I find this easier to read and remember whereas the above seems easy to get wrong or forget but maybe it has other advantages.

Reply:
I think we could modify the above code to instead be:

```
from atomate2.vasp.jobs.base import BaseVaspMaker

bandstructure_flow.update_maker_kwargs(
    {"_set": {"task_document_kwargs->additional_fields": {"mp_id": "mp-xxx"}}},
    dict_mod=True,
    class_filter=BaseVaspMaker,
)
```

Since that way only subclasses of `BaseVaspMaker` will get modified.

That said, it probably makes sense to add a powerup to do this automatically.

Reply:
Okay, I will then add a power-up and showcase that in the above example.

Reply:
👀 Happy Weekend!